test/unit/range/view/view_repeat_n_test.cpp

```diff
@@ -0,0 +1,109 @@
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/core/metafunction/iterator.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/persist.hpp>
+#include <seqan3/range/view/repeat_n.hpp>
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/range/view/take_exactly.hpp>
+#include <seqan3/std/ranges>
+
+using namespace seqan3;
+
+TEST(general, construction)
+{
+    // char
+    char chr{'A'};
+    auto v = view::repeat_n(chr, 4);
+
+    EXPECT_TRUE((std::is_default_constructible_v<decltype(v)>));
+    EXPECT_TRUE((std::is_copy_constructible_v<decltype(v)>));
+    EXPECT_TRUE((std::is_move_constructible_v<decltype(v)>));
+    EXPECT_TRUE((std::is_copy_assignable_v<decltype(v)>));
+    EXPECT_TRUE((std::is_move_assignable_v<decltype(v)>));
+
+    // char const
+    char const chr_const{'A'};
+    auto v_const = view::repeat_n(chr_const, 20);
+
+    EXPECT_TRUE((std::is_default_constructible_v<decltype(v_const)>));
+    EXPECT_TRUE((std::is_copy_constructible_v<decltype(v_const)>));
+    EXPECT_TRUE((std::is_move_constructible_v<decltype(v_const)>));
+    EXPECT_TRUE((std::is_copy_assignable_v<decltype(v_const)>));
+    EXPECT_TRUE((std::is_move_assignable_v<decltype(v_const)>));
+}
+
+TEST(general, concept)
+{
+    char chr{'A'};
+    auto v = view::repeat_n(chr, 10);
+
+    EXPECT_TRUE((std::ranges::Range<decltype(v)>));
+    EXPECT_TRUE((std::ranges::InputRange<decltype(v)>));
+    EXPECT_TRUE((std::ranges::ForwardRange<decltype(v)>));
+    EXPECT_TRUE((std::ranges::BidirectionalRange<decltype(v)>));
+    EXPECT_TRUE((std::ranges::RandomAccessRange<decltype(v)>));
+    EXPECT_FALSE((std::ranges::ContiguousRange<decltype(v)>));
+    EXPECT_TRUE((std::ranges::View<decltype(v)>));
+    EXPECT_TRUE((std::ranges::SizedRange<decltype(v)>));
+    EXPECT_FALSE((std::ranges::CommonRange<decltype(v)>));
+    EXPECT_TRUE((std::ranges::OutputRange<decltype(v), char>));
+}
+
+TEST(view, factory)
+{
+    // const char
+    {
+        char const chr{'X'};
+        auto v = view::repeat_n(chr, 3);
+        EXPECT_EQ(v.size(), 3u);
+        EXPECT_TRUE(std::ranges::equal(v, std::vector<char>{chr, chr, chr}));
+    }
+
+    // string
+    {
+        std::string str{"foobar"};
+        auto v = view::repeat_n(str, 2);
+        EXPECT_EQ(v.size(), 2u);
+        EXPECT_EQ(*v.begin(), str);
+        EXPECT_EQ(v[0], str);
+    }
+
+    // view
+    {
+        auto view = std::string{"foobar"} | view::persist | std::view::take(3);
+        auto v = view::repeat_n(view, 5);
+        EXPECT_TRUE(std::ranges::equal(*v.begin(), view));
+    }
+
+    // combinability
+    {
+        std::string str{"foobar"};
+        auto v = view::repeat_n(str, 5) | view::take_exactly(3);
+        EXPECT_EQ(*v.begin(), str);
+        EXPECT_EQ(std::ranges::size(v), 3u);
```

Comment:
Also here some comments about the view contents would be helpful.

Reply:
I hope the test is more expressive now :)